File: ALGORITHM/programmers/22년9월/구명보트.py
from collections import deque

def solution(people, limit):
    ans = 0
    Q = deque(sorted(people))
    while Q:
        p = Q.pop()
        if Q and p + Q[0] <= limit:
            Q.popleft()
        ans += 1
    return ans

def solution2(people, limit):
    people.sort()
    i = 0
    j = len(people) - 1
    while i < j:
        if people[i] + people[j] <= limit:
            i += 1
        j -= 1
    return len(people) - i

File: ALGORITHM/programmers/2022/09/구명보트.js
function solution1(people, limit) {
  people.sort(function(a, b) {return a-b})
  let i = 0
  let j = people.length- 1
  let ans = 0
  while (i <= j) {
      if (people[i] + people[j] <= limit) {
          i+=1
      }
      j-=1
      ans+=1
  }
  return ans
}

function solution2(people, limit) {
  people.sort(function(a, b) {return a-b})
  let ans = 0
  for (let i = 0, j = people.length - 1; i <= j; j--) {
      if (people[i] + people[j] <= limit) {
          i++
      }
      ans ++
  }
  return ans
}